plugins/inputs/jti_openconfig_telemetry/jti_openconfig_telemetry.go

```diff
@@ -254,8 +254,14 @@ func (m *OpenConfigTelemetry) collectData(
 					&telemetry.SubscriptionRequest{PathList: sensor.pathList})
 				if err != nil {
 					rpcStatus, _ := status.FromError(err)
-					// If service is currently unavailable and may come back later, retry
-					if rpcStatus.Code() != codes.Unavailable {
+					if rpcStatus.Code() == codes.Unauthenticated {
+						if m.Username != "" && m.Password != "" && m.ClientID != "" {
+							if err := m.authenticate(ctx, grpcServer, grpcClientConn); err != nil {
+								acc.AddError(fmt.Errorf("could not re-authenticate: %w", err))
+							}
```

Comment:
Shouldn't we `continue` on success to retrigger subscription (the function in line 253)? Otherwise we might `return` at the end of the iffery (line 275) and gain nothing...

This whole construct concerns me a bit as I think the `RetryDelay` handling (line270) is flawed. The GRPC code does reconnects with its default strategy as necessary and we have no idea on the state changes. So if the connection is lost and regained we run into this unauthenticated branch. However, if the `RetryDelay` is zero, we  still exit on line 275. We could circumvent this with a `continue`  statement on login success, as suggested above, but then we might ignore the given delay. I think both is wrong, especially as we might end-up in a fast subscribe-reconnect loop if the connection is really flaky and the connection is always lost directly _after_ we successfully logged-in...

Reply:
thank you for catching the missing continue. I've added that + a sleep based on the retry delay. Let me continue to think about how to handle the flaky and/or subscribe-reconnect loop.